threads/src/main/java/org/jboss/as/threads/EnhancedQueueExecutorResourceDefinition.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2019, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.threads;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.PersistentResourceDefinition;
+import org.jboss.as.controller.ReadResourceNameOperationStepHandler;
+import org.jboss.as.controller.SimpleResourceDefinition;
+import org.jboss.as.controller.capability.RuntimeCapability;
+import org.jboss.as.controller.registry.ManagementResourceRegistration;
+import org.jboss.msc.service.ServiceName;
+
+import static org.jboss.as.threads.CommonAttributes.ENHANCED_QUEUE_THREAD_POOL;
+
+/**
+ * {@link org.jboss.as.controller.ResourceDefinition} for an {@code org.jboss.threads.EnhancedQueueExecutor} resource.
+ *
+ */
+public class EnhancedQueueExecutorResourceDefinition extends PersistentResourceDefinition {
+    private final EnhancedQueueExecutorWriteAttributeHandler writeAttributeHandler;
+    private final EnhancedQueueExecutorMetricsHandler metricsHandler;
+
+    private final boolean registerRuntimeOnly;
+    public static final RuntimeCapability<Void> CAPABILITY =
```

Comment:
Also, ManagedEnhancedQueueExecutor is not public.  That would be ok if the only consumers of the service exposed by this capability were in this package, but that's not the case here.

It's possible this could just be ExecutorService, though. The public methods in ManagedEnhancedQueueExecutor are just a bunch of getters for config values or metrics, so it's quite possible external users of the capability have no need for any of that API. The users of that API are the management classes in this PR.

Reply:
@bstansberry thanks for reviewing.  I've made the following changes based on your comments.

Changed to use dynamically-named capability.  In `create` method that no capability is passed in, create the capability using `ThreadsServices.createCapability(...)` with `type` param value.

Changed service value from `ManagedEnhancedQueueExecutor` to just `ExecutorService`, and so `ManagedEnhancedQueueExecutor` remains non-public.

Changed internal classes and methods to be non-public.  `EnhancedQueueExecutorResourceDefinition` remains public.

Rearranged imports following project convention.